File: rfidreader.server/src/main/kotlin/com/rfidreader/models/Student.kt
package com.rfidreader.models

import jakarta.persistence.*

@Suppress("JpaObjectClassSignatureInspection")
@Entity
@Table(name = "Students")
data class Student (
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long? = null,
    var name: String,
    var surname: String,
    var patronymic: String,
    var rfidCode: String,
) {
    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "group_id")
    var group: Group? = null

    @OneToMany(mappedBy = "student", cascade = [CascadeType.ALL])
    var attendances: MutableList<Attendance> = mutableListOf()
}

File: rfidreader.desktop/src/renderer/src/models/student.ts
import { IGroupInfo } from "./group"

export interface IStudentInfo {
    id: number
    name: string
    surname: string
    patronymic: string
    rfidCode: string
    group: IGroupInfo 
}
export interface INewStudent {
    name: string
    surname: string
    patronymic: string
    rfidCode: string
    groupId: number
}
export interface IUpdateStudent {
    id: number
    name: string
    surname: string
    patronymic: string
    groupId: number
    rfidCode: string
}